src/basic.h

```diff
@@ -61,8 +61,10 @@ class Symbol;
 */
 
 enum TypeID {
-    SYMBOL, MUL, ADD, POW, LOG,
-    NUMBER, INTEGER, RATIONAL, COMPLEX, CONSTANT,
+    INTEGER, RATIONAL, COMPLEX, REAL_DOUBLE, NUMBER,
+    // 'NUMBER' returns the number of subclasses of Number.
+    // All subclasses of Number must be added before it.
```

Comment:
How about renaming `NUMBER` to `TypeID_Number_Count`, to make it clear what it does?